controllers/mentor.go

```diff
@@ -105,3 +126,84 @@ func FetchAllMentors(w http.ResponseWriter, r *http.Request) {
 
 	utils.RespondWithJson(r, w, mentors)
 }
+
+// /mentor/dashboard/ functions
+
+func CreateMentorDashboard(mentor models.Mentor, db *gorm.DB) MentorDashboard {
+	var projects []models.Project
+	var projectsInfo []ProjectInfo
+	var students []StudentDashboard
+
+	db.Table("projects").
+		Where("mentor_id = ? OR secondary_mentor_id = ?", mentor.ID, mentor.ID).
+		Preload("Mentor").Preload("SecondaryMentor").
+		Find(&projects)
```

Comment:
It would be better if you just select the necessary fields here. (See the `.Select()`) function.

Reply:
You can also directly put the information into a ProjectInfo struct array. See this https://github.com/kossiitkgp/KWoC-Backend/blob/124b8783cacf1811d69b67f3aaf3da34874fa5da/controllers/student.go#L187-L191